### ISSUE TITLE
Archiving and marking as read actions in N1 not being reflected on Gmail 

### ISSUE THREAD

**Issue by markbao** [nylas/N1#2059]
Don't have a way to reproduce this, but N1 isn't syncing correctly with 3 of 4 of my accounts. I am supposedly at Inbox Zero in N1, but in the inboxes for 3 out of 4 of my accounts, I have 1, 1, and 8 messages in my inbox.

**Comment by markbao** [comment]
Continuing to have this issue. Archiving messages sometimes does not actually pull them out of the inbox on the server, though they disappear from N1's inbox.

**Comment by markbao** [comment]
Related to #2060

**Comment by dvdfox** [comment]
I have this problem in reverse. Unibox mail client and the webmail portal correctly show that I have 1 email in my Inbox. N1 shows a bunch of mail that was already archived by my other email client. It never updates. It's been over 24 hours for some of these messages.

![screen shot 2016-04-28 at 2 49 59 pm](https://cloud.githubusercontent.com/assets/6954139/14897873/e1b17ede-0d51-11e6-88e2-d02ffaf0ad74.png)
![screen shot 2016-04-28 at 2 50 10 pm](https://cloud.githubusercontent.com/assets/6954139/14897874/e1d934ba-0d51-11e6-99cc-71dee27733dd.png)
![screen shot 2016-04-28 at 2 50 30 pm](https://cloud.githubusercontent.com/assets/6954139/14897875/e1dca5c8-0d51-11e6-85d6-ef3d15737acd.png)

**Comment by esttorhe** [comment]
Instead of creating a new issue i'm just going to add that this is happening for me as well.

1 of my accounts is showing me an email from Dec 14th, 2015 as my only pending email for that account.

While all other email clients (even Gmail) shows me 2 emails (none of them related to this one Nylas is showing) and I can't sync new emails to that account :/

Another account that's not syncing is showing Inbox 0 on Nylas while I have 8 emails on that Inbox.

Something to notice is that both accounts have 2-FA enabled

**Comment by markbao** [comment]
Issue persists in Version 0.4.33-e8f137e (0.4.33-e8f137e). At this version things seem to be pretty out of whack.

`@columbia` address:
- Before: 7 messages in Gmail, 8 messages in N1. (1 message shown in N1 that is not in the Gmail inbox)
- Archived first message.
- After: 7 messages in Gmail (first message not archived), 7 messages in N1. Original email has disappeared, but the "extra" message that is in N1 but not Gmail is still there.

`@decisionsciences` address:
- Before: 8 messages in Gmail, 8 messages in N1 (all OK here)
- Archived 3 messages
- After: 8 messages in Gmail (all messages failed to archive), 5 messages in N1.

**Comment by adambrault** [comment]
I think I'm seeing the same thing. Archiving and read states in N1 aren't showing up in Gmail.

**Comment by esttorhe** [comment]
Is this something that its being worked on?

I really love `Nylas` but without proper syncing its unusable for me.

I'll try to give a take at this issue and if I can submit a `PR` or something with a patch i'll do 😄

**Comment by markbao** [comment]
Update: Archiving is no longer working now.

Version 0.4.33-e8f137e (0.4.33-e8f137e)

**Comment by markbao** [comment]
Related to #2172 #1247

**Comment by jstejada** [comment]
hi @markbao , thanks for the report. Can you show me what the activity window shows when you try to archive? You can open it via `Developer -> Open Activity Window`

Thanks!

**Comment by markbao** [comment]
Thanks @jstejada!

Nothing out of the ordinary it seems (IDs redacted). The archived messages are still shown in the Gmail inbox, however.

![asdf](https://cloud.githubusercontent.com/assets/27594/15191914/d157b4f8-1784-11e6-8114-b8a718fb8702.png)

**Comment by markbao** [comment]
Let me know if you want me to email you the unredacted version and a safe way to do that (not sure if the username in front of :@api.nylas.com is a secret identifier or not)

**Comment by jstejada** [comment]
hmmm, no need. so no failures anywhere? can you send me the email of the account your having trouble with on the nylas community slack?

**Comment by markbao** [comment]
Done.

**Comment by jstejada** [comment]
Thanks! This might have been due to a minor sync outage we had today -- your accounts should catch up with sync within today. @esttorhe , @adambrault , let me know if you continue to have problems. I'll leave this open for now

**Comment by esttorhe** [comment]
I'll keep you posted @jstejada although I've been experiencing this issue since like a week ago or so.

Already launched `Nylas` again; will let you know tomorrow.

**Comment by esttorhe** [comment]
@jstejada FYI my inbox is still out of sync.

Its been over 23 hours since I last opened `Nylas` and now none of my accounts display the emails correctly.

I do get new emails on accounts that do not have 2-FA but none of my inbox are showing emails whilst on other clients and on the web client I can see at least 6 emails on each account…

:/ sadly the behavior is worst than before

**Comment by jstejada** [comment]
hey @esttorhe , thanks for checking back. Are you on the community slack by any chance? Could you send me the email address for your account there?

**Comment by esttorhe** [comment]
I am not ATM.

Care to either add me or show me the signup link and I'd gladly join 😄

**Comment by jstejada** [comment]
you can join here: http://slack-invite.nylas.com/
you can also just send me your account info to support@nylas

**Comment by Smona** [comment]
Just started using Nylas today and I'm in love. However, I'm experiencing the same issue. My output window is also showing sync successes, but emails are neither gettting marked as read or archived in gmail.

**Comment by esttorhe** [comment]
@jstejada done; thanks 🙇

**Comment by Mobilizr** [comment]
Same issue here. 

In Nylas N1 my Inbox shows "Hooray! You're done." - i.e. zero messages.

But when I login to gmail web or use any mobile apps on my phone, the messages I have been deleting or archiving from the Inbox in N1 are still sitting there. And it is not just that they are still in the Inbox - they are also showing as "unread" which is also not correct...

**Comment by jstejada** [comment]
hi @Mobilizr and all, I believe I've narrowed down the issue, and it seems to have been introduced with an incorrect fix for #2049. (also related to #2036). This will be fixed in the next patch release which will go out this week

for anyone who is having issues, please try the following in the meantime: 

> As a temporary fix, you can close N1, remove any cursor fields you have inside your `config.cson` and then restart the app. This should fix the issue for now. The cursor field should be located here in the file:
```
...
nylas:
  <accountId>:
    cursor: "..."
...
```

Please let me know if this works

**Comment by Smona** [comment]
Good news! Looks like archiving and marking read are syncing for me now. Didn't even apply the fix you posted.

**Comment by markbao** [comment]
Thanks Juan. I applied the cursor fix but unfortunately messages are still not being marked as read or archived. I'll wait for the patch release to readd my accounts. Fingers crossed!

**Comment by markbao** [comment]
Thanks Juan. I've tried the cursor fix above, but unfortunately I'm still having the issue. I removed my accounts, fully wiped N1 (removed all app storage, preferences, etc. with AppZapper), and re-installed and re-added my accounts, but continue to have the issue (messages are still not archiving).

**Comment by esttorhe** [comment]
For what is worth here's my experience with this issue.

A **LOOOOONG** time ago I tried `N1` but wasn't working for me for one reason or another; I uninstalled and that's it.
I tried `N1` a couple more times always with the same result.

Last week I decided to give it another go and was experiencing the issue described here.
Juan took some time to try and help me but after the `cursors` fix my app was still behaving incorrectly.


In the end deleting `~/.nylas` completely and re adding my accounts was the only solution that actually worked.

Looks like at some point (probably on my first try) there was some broken configuration that stayed on my folder and every time I «tried» a new i was actually pilling over the broken config.

Not saying is a perfect solution or even a solution for most of you but in my case that was the path to make `N1` usable for me

**Comment by jstejada** [comment]
Thanks for sharing @esttorhe! And yes, if the cursor fix does not work for you, the way to completely wipe out your data is by completely removing you `~/.nylas` folder. If anyone in this thread is still having issues after those fixes please let me know. Thanks! And sorry for the trouble (cc @markbao, @dvdfox, @Mobilizr, @adambrault )

**Comment by markbao** [comment]
I completely removed my `~/.nylas` folder, restarted the app, and got this screen:

![image](https://cloud.githubusercontent.com/assets/27594/15255742/f0a09c34-190b-11e6-8a61-9ace78505dd8.png)

After restarting the app another time, it sent me to the correct set up page. Minor bug maybe.

I set up my accounts again and tested again, but I'm still having the same issue as before - messages not being marked read/archived.

**Comment by jstejada** [comment]
thanks @markbao , looking into your issue now

**Comment by markbao** [comment]
Not sure if it's useful, but I've noticed that archiving emails on Gmail inbox successfully hides them from the N1 inbox, so it's just the one-way N1 to Gmail archiving that's not working. Adding labels also works correctly.

This is weird. On my main account (`mark@markbao`), I did some combination of archiving in Gmail inbox, then archiving in N1, and starring / adding labels to items. Now things seem to be archiving correctly. I'll continue playing with it.

**Comment by adambrault** [comment]
That matches my experience @markbao

**Comment by jstejada** [comment]
hi @adambrault, can you send me your email address via email or slack? http://slack-invite.nylas.com/

**Comment by adambrault** [comment]
@jstejada feel free to use the one on my GH account

**Comment by jstejada** [comment]
@adambrault, thanks, I just checked your account for that email and it seems to be syncing correctly. are your archiving and unread states still not showing up in gmail?

**Comment by jstejada** [comment]
@dvdfox, @Mobilizr, let me know if youa re still having issues. if so, send me your email addresses. Thanks! -- sorry for the hassle

**Comment by dvdfox** [comment]
My email is dfox@auras.com. I’ve been on vacation so I haven’t even launched Nylas for over two weeks. Launching it now, without taking any of the troubleshooting steps you’ve suggested shows 9 unread email in my Inbox. In Unibox, I have only one already read message from you. So it’s still not working, but, like I said, I haven’t had time to do any of the fixes you suggested. 




  
  David FoxAURAS Design | 8435 Georgia Ave., Third Floor | Silver Spring, MD 20910 | 301-587-4300

**Comment by dvdfox** [comment]
Oh, I guess I also have it set up for my gmail too: dvdfox39@gmail.com




  
  David FoxAURAS Design | 8435 Georgia Ave., Third Floor | Silver Spring, MD 20910 | 301-587-4300

**Comment by adambrault** [comment]
@jstejada You'll note below the difference between the state in Nylas vs. Gmail:

<img width="142" alt="screenshot 2016-05-16 12 50 18" src="https://cloud.githubusercontent.com/assets/110551/15301867/c82cd410-1b64-11e6-8f85-2c4d3559abd8.png">

<img width="365" alt="screenshot 2016-05-16 12 51 37" src="https://cloud.githubusercontent.com/assets/110551/15301908/ead1faae-1b64-11e6-9b0e-698a60b84b03.png">

Gmail shows an email I've read and archived still in the inbox, unread, and shows other emails that I've archived still in the inbox. (The other emails are read because I read them on my mobile phone.)

**Comment by adambrault** [comment]
(This is with the latest version — 0.4.38-27d21d1)

**Comment by jstejada** [comment]
thanks @adambrault !

**Comment by jstejada** [comment]
So we've confirmed that an unusually high number of actions are failing to be transmitted to the mail provider. We are currently looking into this and will post any updates in this issue. Thanks for your help debugging! and sorry for the hassle! cc @kav-ya

**Comment by nienque** [comment]
Just want to pitch in and let you know I've been having the same problem. I think it's 'reply + archive' for me that's the problem, all emails that aren't properly moved to the archive for me were ones I 'reply + archive'd.

**Comment by pattydaniels** [comment]
I also am having this issue. Unfortunately, I've had to stop using Nylas N1 altogether because of this. Hope it gets fixed soon so I can return to it.

**Comment by adambrault** [comment]
Same. Just quit using it over this, too.

It's a very good product otherwise!

**Comment by glen3dot1** [comment]
@adambrault & @pattydaniels — the problem I was having was due to a temporary sync problem on *some* of the Nylas servers. I think that's what Juan was referring to. Anyway, I haven't had any troubles since.

**Comment by payammeyer** [comment]
I definitely still have this issue with my Gmail accounts, both personal and business.

**Comment by Kashura** [comment]
@jstejada Thanks for Merging. I searched but didn't find this one. 
When you say "unusually high number of actions", what would you consider as a high number? I have 5 email accounts, and I wouldn't say that gong through my inbox is a high number :-)

**Comment by Mobilizr** [comment]
I thought this issue had resolved itself, but @ today it is well and truly back again. In N1 I just went through all my mail and got down to "Zero Inbox". Then I jumped onto my iPhone a few minutes later - it was like I hadn't done anything at all. Emails I had read in N1 were still "unread" and all the mail I had sorted/deleted/archived were still sitting there in the Inbox.

I am loving N1, but this issue is not going away, and it is seriously impacting on my workflow and my productivity at this point. I am going to have to stop using it altogether until this issue is resolved. I hope it can be. I have never seen this problem before in other mail clients, and I have tried them all.

All the best with getting it sorted.

**Comment by Foof** [comment]
Had the same issue with archiving in Nylas not reflecting in my Gmail account ever since I first downloaded it over a month ago. Assumed it was a temporary bug, but it still isn't working properly. Will stop using it for now and check back later to see if this issue has been resolved.

**Comment by pattydaniels** [comment]
Any update on this? I'd love to get back to using Nylas N1.

**Comment by markbao** [comment]
Unfortunately, myself and @lleger are seeing this issue again. Marking as read and Archive is not working again.

**Comment by lleger** [comment]
Yes, what @markbao. It's made N1 unusable, I've stopped using it for the time being. I barely want to deal with my inbox once, please don't make me do it twice!

**Comment by Kashura** [comment]
@pattydaniels Unfortunately, no :-(

**Comment by MaximOrlovsky** [comment]
same issue with read emails and archive. Nylas N1 also does not delete emails in gmail after I deleted them in Nylas N1.

**Comment by alexjcollins** [comment]
Likewise re Gmail. Seems ok with iCloud (if a little slow)

**Comment by urfolomeus** [comment]
I'm also seeing this issue. I've deleted all my accounts and my `.nylas` folder, then reinstated them, but still getting the issue.

On Mac OSX 10.11.5, N1 0.4.45-7637265.

**Comment by bartlomn** [comment]
Hi @NylasTeam, any update on the resolution status of this issue?

**Comment by FlorianBrinkmann** [comment]
@jstejada Thanks for merging my issue with this one. Would be great to get this solved quickly, because like @payammeyer said it's really annoying …

**Comment by JherezTaylor** [comment]
Can confirm. Started for me with the 0.4.45.7..65 update

**Comment by Pete5x5** [comment]
I've been having this issue for at least a month.  Sent comments and details to a couple other threads that were merged into this one.  Very frustrating issue.

**Comment by MaximOrlovsky** [comment]
Could anybody suggest which version of Nylas N1 have correct sync with
Gmail? I'm ready to downgrade.

**Comment by markbao** [comment]
This issue is back for me for the fourth time. Not sure if related, but I upgraded last night to 0.4.45-7637265.

**Comment by lleger** [comment]
Confirmed it's still present on the latest version for me as well.

**Comment by nienque** [comment]
Same here, updated to the latest version, tried archiving some emails and Gmail web interface doesn't reflect the changes. Same thing for iCloud though, it's not just Gmail for me.

**Comment by victorBaro** [comment]
Same here in latest version

**Comment by raymurillo** [comment]
Updated today and I'm still experiencing this as well. It's been an issue since I've first tried Nylas. I like the product, but this is frustrating.

**Comment by azazelpy** [comment]
Updated today and go to the pro version, same issue with Gmail and outlook. 
I try others email clients at the same time without problem

**Comment by piavgh** [comment]
The same issue happened to me for the last 3 to 4 days. The issue was not fixed

**Comment by AmberHaccou** [comment]
I really like Nylas, but also experiencing this issue. I can't work like this :(

**Comment by FlorianBrinkmann** [comment]
I tested a custom imap account with just a few mails (less than 20) and there it works fine. My other custom IMAP accounts which doesn’t sync correctly have much more messages. I tried a downgrade up to 0.4.25 but it didn’t work either. So I think it’s a problem on server side. Have to switch to another mail client until the issue is fixed -.-

**Comment by nashcheez** [comment]
Yes syncing of email is out of order. Also, when I am sending an email, the corresponding email is not appearing in the Sent email folder. I though this might be an issue when the 'Link tracking' and 'Open tracking' opens are enabled, so I disabled them, and it worked just fine for the next few messages.
But then again after a while, it stopped syncing.
I have tried almost all email clients that have existed for the mac platform, and I have gotten myself to really like N1. Can't persist with it though if these issues aren't resolved.
Version: 0.4.45-7637265

**Comment by davidwolski** [comment]
Same here on Gmail account. The archiving the mail in Nylas makes it disappear from the Web Gmail Inbox and appear in the All Mail folder with an Inbox label.

**Comment by RobertFreitas** [comment]
I'm having the same syncing issues. logging into my basic webmail client, I can see any of the actions I've performed in Nylas don't transfer. as a result, those actions also don't show up in the mail app on my phone. has there been any update at all on this? it makes the software totally useless for me, which is a massive let down, as I love the interface and tools.

**Comment by logandavis** [comment]
Hi all - Logan from Nylas here. This is our platform team's #1 priority at the moment - we're aware of the issues and have multiple backend engineers working on it this afternoon. Hopefully we'll have these issues resolved quickly; I'll let you know here when they do.

**Comment by payammeyer** [comment]
+1 to Logan.

**Comment by jstejada** [comment]
We’ve also opened a statuspage incident for this issue, you can track it here: http://status.nylas.com/incidents/9fcs2r3bbk32.

**Comment by osakkers** [comment]
It seems that the issue is resolved for me.

**Comment by davidwolski** [comment]
Same here, syncing works fine at the
moment.![](https://link.nylas.com/open/d9wwk1krjd38al8yqcj754wl9/local-
7499241e-
2c12?r=cmVwbHkrMDAzYmU1NTE1MWE0YzI2ZjhlYmIyNTBlMjEyNDk5ZDRkZDkzNWM4M2NiMzdmN2I0OTJjZjAwMDAwMDAxMTM3YmNhMjU5MmExNjljZTA4ZmZiY2U5QHJlcGx5LmdpdGh1Yi5jb20=)

On Jun 17 2016, at 10:24 am, osakkers &lt;notifications@github.com&gt; wrote:  

> It seems that the issue is resolved for me.

>

> —  
> You are receiving this because you commented.  
> Reply to this email directly, [view it on
GitHub](https://github.com/nylas/N1/issues/2059#issuecomment-226782752&r=cmVwbHkrMDAzYmU1NTE1MWE0YzI2ZjhlYmIyNTBlMjEyNDk5ZDRkZDkzNWM4M2NiMzdmN2I0OTJjZjAwMDAwMDAxMTM3YmNhMjU5MmExNjljZTA4ZmZiY2U5QHJlcGx5LmdpdGh1Yi5jb20=),
or [mute the
thread](https://github.com/notifications/unsubscribe/ADvlUZ20eik3xLFwDrrZi1ip9SsD4K_hks5qMq4lgaJpZM4IPWZs&r=cmVwbHkrMDAzYmU1NTE1MWE0YzI2ZjhlYmIyNTBlMjEyNDk5ZDRkZDkzNWM4M2NiMzdmN2I0OTJjZjAwMDAwMDAxMTM3YmNhMjU5MmExNjljZTA4ZmZiY2U5QHJlcGx5LmdpdGh1Yi5jb20=).![](https://github.com/notifications/beacon
/ADvlUc4EtEu-Gd-JUvjiCynCFmRC--41ks5qMq4lgaJpZM4IPWZs.gif)

**Comment by nashcheez** [comment]
Yeah, issues seems to be resolved for the time being. 
Cheers to the Nylas team. 👍

**Comment by urfolomeus** [comment]
Back to normal for me too! \o/

Thanks folks :)

**Comment by alexjcollins** [comment]
The issue still persists for me :(

**Comment by azazelpy** [comment]
now work for me.

@alexjcollins try delete and add again your accounts

**Comment by alexjcollins** [comment]
@azazelpy yeah tried that (including deleting ~/.nylas) - hopefully it'll sort itself out shortly

**Comment by ptrkcsk** [comment]
Fired up N1 this morning and it looks like the issue has been resolved. Thanks!

**Comment by bartlomn** [comment]
Resolved for me as well. Well done!

**Comment by logandavis** [comment]
Hey all,

I'm not one of the engineers working on this issue - they stayed up all night last night and are still very busy working! - but I understand that most of the problem was caused by huge backlogs of actions (e.g. mark as unread, archive) that needed to be synced from N1 back to the mail provider coming from individual accounts. This can happen, for example, when you apply a mail rule to your entire inbox. We think we've managed to resolve the problem for about 98% of accounts, but the 2% of accounts that were causing the backlog (through no fault of the users) still have a huge backlog of actions to work through and are still experiencing sync issues. (The backlog of pending actions that we've registered but our overloaded servers haven't been able to work through yet is down from around 500,000 at its peak to about 150,000 now, most of which are coming from only 50 N1 accounts.)

A more detailed postmortem will likely follow - just thought I'd keep y'all in the loop. Glad to hear the issue has resolved for most of you, and for those of you who it hasn't resolved for, know that we're working on it!

**Comment by urfolomeus** [comment]
Thanks for the update, @logandavis! Appreciate the time you folks are putting in on this. Loving N1 <3

**Comment by Eadword** [comment]
Thunderbird's days are now numbered. You guys are great.

**Comment by alexjcollins** [comment]
Yeah, appreciate the update @logandavis

**Comment by payammeyer** [comment]
Yep thanks so much for the update AND looking into this problem. It looks like I'm among the 2%, still seeing the issue on and off.

**Comment by markbao** [comment]
You all are amazing. Seems fixed for me, hopefully not too long for the others. Thank you!

**Comment by lleger** [comment]
Seems to be working for me as well! 🎉 💯 👏

**Comment by jkudish** [comment]
Working here as well. Thanks so much for addressing this so quickly!

**Comment by khamidou** [comment]
Hey there --- we've just shipped a fix to make applying changes fairer for all users (commit here for the curious: https://github.com/nylas/sync-engine/commit/2c3f403e6d7a7a6fc67fa26a5cba10ae5c820fd6). Please let me know if you're still encountering this issue.

Thanks,

**Comment by grinich** [comment]
We've finished processing the queue of actions and this issue should be resolved. Is anyone still seeing issues?

**Comment by alexjcollins** [comment]
@grinich working great now, thanks!

**Comment by Hudell** [comment]
I still had to organize the existing messages manually, but it's working great for the new ones. Thanks!

**Comment by grinich** [comment]
OK awesome. Gonna close this issue! 🎉✌️

**Comment by jstejada** [comment]
#781
